statsmodels/tsa/statespace/tests/test_mlemodel.py

```diff
@@ -669,7 +669,7 @@ def test_pandas_endog():
     mod.filter([])
 
     # Example : pandas.Series, string datatype
-    endog = pd.Series(['a'], index=dates)
+    endog = pd.Series(['a', 'b'], index=dates)
     # raises error due to direct type casting check in Statsmodels base classes
     assert_raises(ValueError, check_endog, endog, **kwargs)
```

Comment:
The change here changes the reason for the ValueError.
@ChadFulton  What's the point of this assert?

Reply:
I don't think it changes the reason for the ValueError. `dates` is an index of length 2, and I guess in older Pandas, it would "broadcast" the list `['a']` to `['a', 'a']` to make it fit the index. In newer Pandas (Pandas >= 0.23?) I guess it raises an error.

The reason for the `ValueError` is to check that if you pass in strings, Statsmodels will complain (in `/base/data.py:76`)